packages/core/src/transport/batch.ts

```diff
@@ -149,15 +149,15 @@ export class Batch {
        */
       addEventListener(document, DOM_EVENT.VISIBILITY_CHANGE, () => {
         if (document.visibilityState === 'hidden') {
-          this.flush()
+          this.flush('visibility_hidden')
         }
       })
       /**
        * Safari does not support yet to send a request during:
        * - a visibility change during doc unload (cf: https://bugs.webkit.org/show_bug.cgi?id=194897)
        * - a page hide transition (cf: https://bugs.webkit.org/show_bug.cgi?id=188329)
        */
-      addEventListener(window, DOM_EVENT.BEFORE_UNLOAD, () => this.flush())
+      addEventListener(window, DOM_EVENT.BEFORE_UNLOAD, () => this.flush('before_unload'))
```

Comment:
It's because I mainly want to know when we are loosing data:
- before_unload/visibility_hidden => data loss
- other cases => the fallback should works.

Reply:
The PR description mention collision but IIUC we are collecting all failed requests, with every possible reason on replay but with only a subset of reason on logs/rum.
Here, you are mentioning data loss but we could have collisions without data loss, no?
I am a bit confused on what we are trying to achieve and how we plan to analyze the results out of it 😕

Reply:
I updated the description